test/gets.js

```diff
@@ -20,59 +21,76 @@ describe('gets', function() {
            errorEq(TypeError,
                    'Invalid value\n' +
                    '\n' +
-                   'gets :: Accessible a => TypeRep -> Array String -> a -> Maybe b\n' +
-                   '                        ^^^^^^^\n' +
-                   '                           1\n' +
+                   'gets :: Accessible a => (Any -> Boolean) -> Array String -> a -> Maybe c\n' +
+                   '                        ^^^^^^^^^^^^^^^^\n' +
+                   '                               1\n' +
                    '\n' +
                    '1)  [1, 2, 3] :: Array Number, Array FiniteNumber, Array NonZeroFiniteNumber, Array Integer, Array ValidNumber\n' +
                    '\n' +
-                   'The value at position 1 is not a member of ‘TypeRep’.\n'));
+                   'The value at position 1 is not a member of ‘Any -> Boolean’.\n'));
 
-    throws(function() { S.gets(Number, null); },
+    throws(function() { S.gets(String, ['x'], {x: 'a'}); },
            errorEq(TypeError,
                    'Invalid value\n' +
                    '\n' +
-                   'gets :: Accessible a => TypeRep -> Array String -> a -> Maybe b\n' +
-                   '                                   ^^^^^^^^^^^^\n' +
-                   '                                        1\n' +
+                   'gets :: Accessible a => (Any -> Boolean) -> Array String -> a -> Maybe c\n' +
+                   '                                ^^^^^^^\n' +
+                   '                                   1\n' +
+                   '\n' +
+                   '1)  "a" :: String\n' +
+                   '\n' +
+                   'The value at position 1 is not a member of ‘Boolean’.\n'));
```

Comment:
I added these additional assertions in the unit tests to make sure the cases of `S.get(Array, 'x', {x: 1}) -> Just(1)` are resolved once and for all.